icepack.setup

```diff
@@ -876,10 +891,18 @@ cp ${rundir}/compile/*.{gcno,gcda} ${testname_base}/codecov_output/
 
 EOF
 
+cat >> ${tsdir}/report_lcov.csh << EOF
+lcov --gcov-tool gcov -c -d ${rundir}/compile -o ${testname_base}/lcov.info
+if (-s ${testname_base}/lcov.info) then
+  set lcovalist = "\${lcovalist} -a ${testname_base}/lcov.info"
+endif
+
+EOF
+
 cat >> ${tsdir}/suite.run << EOF
 cd ${testname_base}
 ./icepack.build
-./icepack.submit
+./icepack.submit | tee -a ../suite.jobs
```

Comment:
what does this do?

Reply:
This takes the output of the submit script and redirects it to the file suite.jobs in addition to going to standard out.  The tee allows you to redirect output to two places.  What we're doing is creating a file with all the jobs ids.  Once all the jobs are submitted, this file is parsed and each job is checked that it is no longer in the queue before doing the final step which may be to report the results or do the code coverage analysis.  The key is that we need to allow all the jobs to complete before doing the final reporting (if --report or --coverage) is chosen and we capture the jobids with the tee.

At the bottom of icepack.setup (cice.setup is the same), we have

```
  if ($report == 1) then
    echo "Reporting results"
    ./poll_queue.csh
    ./results.csh
    ./report_results.csh
  endif
  if ($coverage == 1) then
    echo "Generating coverage reports"
    ./poll_queue.csh
    ./report_lcov.csh
#    ./report_codecov.csh
  endif
```

poll_queue.csh is the script that checks whether the jobs are done.  We only do that checking if --report or --coverage is set.  Finally, just fyi, suite.jobs looks like

2865298.chadmin1.ib0.cheyenne.ucar.edu
2865301.chadmin1.ib0.cheyenne.ucar.edu
2865305.chadmin1.ib0.cheyenne.ucar.edu
...

and what is surprising is that different machines and queuing systems seem to use more or less the same syntax for jobids, so we can simply parse the jobids from each line and verify that that job is no longer in the queue.  This implementation has always seemed a little dicey to me, but it has held up.

Reply:
Cool, thanks for the explanation.  I don't think the scripts work quite the same way on badger (e.g. I get lots of FAILs when runs are getting set up, rather than PENDs, until the suite is completely finished), but I usually don't invoke --report on the command line and haven't worried about it.  If it becomes a problem on other platforms then maybe we can fix it here too.